src/logcollector/read_win_event_channel.c

```diff
@@ -487,64 +496,65 @@ void send_channel_event(EVT_HANDLE evt, os_channel *channel)
     }
     xml_event = convert_windows_string((LPCWSTR) properties_values);
     
-    find_prov = strstr(xml_event, "Provider Name=");
-  
-    if(find_prov){
-        beg_prov = strchr(find_prov, '\'');
-        if(beg_prov){
-            end_prov = strchr(beg_prov+1, '\'');
-
-            if (end_prov){
-                num = end_prov - beg_prov - 1;
-
-                if(num > OS_MAXSTR - 1){
-                    mwarn("The event message has exceeded the maximum size.");
-                    goto cleanup;
+    if (xml_event) {
```

Comment:
Instead of encapsulating the entire block of code in an `if` statement, include the next check before this line:

```
if (!xml_event) {
    goto clean_up;
}
```

It is equivalent and more clear.